sklearn/tree/_classes.py

```diff
@@ -266,18 +267,7 @@ def fit(self, X, y, sample_weight=None, check_input=True,
                               "or larger than 1").format(max_leaf_nodes))
 
         if sample_weight is not None:
-            if (getattr(sample_weight, "dtype", None) != DOUBLE or
-                    not sample_weight.flags.contiguous):
-                sample_weight = np.ascontiguousarray(
-                    sample_weight, dtype=DOUBLE)
-            if len(sample_weight.shape) > 1:
-                raise ValueError("Sample weights array has more "
-                                 "than one dimension: %d" %
-                                 len(sample_weight.shape))
-            if len(sample_weight) != n_samples:
-                raise ValueError("Number of weights=%d does not match "
-                                 "number of samples=%d" %
-                                 (len(sample_weight), n_samples))
+            sample_weight = _check_sample_weight(sample_weight, X, np.float64)
```

Comment:
Why not keeping `dtype=DOUBLE` ?

Reply:
The `_check_sample_weight` only supports `np.float32` and `np.float64`.

Reply:
@eickenberg

Reply:
`DOUBLE = np.float64`.
While I agree this is equivalent to write `np.float64`, using `DOUBLE` makes it clear that it must be the same type as what is used in the tree cython code. So I would slightly prefer keeping `dtype=DOUBLE`.

Reply:
(this is very much nitpicking)

Reply:
oh, if `DOUBLE` is set to be/mean `np.float64`, then please go with that for consistency and ease of change in the future

Reply:
Great! We just updated the method and added a more specific test assertion as per our earlier offline discussion.